src/providers/codeforces.cr

```diff
@@ -0,0 +1,37 @@
+class Codeforces < SiteStats
+  @@url_regex = Regex.new("https?://codeforces[.]com/profile/(?<username>\\w+)")
```

Comment:
I would add `(www[.])?` since https://www.codeforces.com/profile/rafarafa is also a valid codeforces profile url.